lib/problem3.rb

```diff
@@ -0,0 +1,12 @@
+def problem3(values)
+  incremental_product = 1
+  index_counter = 1
+  array_product = 1
+  while value = values.shift do
+    array_product *= value
```

Comment:
Please check: https://www.calculatorsoup.com/calculators/discretemathematics/factorials.php for calculating factorials. Given you have N numbers from 1 to N. What is the biggest N you can have for your algorithm to run correctly?